File: AppData/Mods/BUILT-IN/_lua/LuaScript/Uzil/Achv.lua
local this = {};

--[[ 是否 已解鎖 ]]
function this.isUnlock (id)
	return UZAPI.Achv.IsUnlock(id);
end

--[[ 設置 進度 ]]
function this.getStat (id)
	return UZAPI.Achv.GetStat(id);
end

--[[ 設置 進度 ]]
function this.setStat (id, progress_int)
	UZAPI.Achv.SetStat(id, progress_int);
end

--[[ 設置 完成 ]]
function this.setDone (id)
	UZAPI.Achv.SetDone(id);
end

return this;

File: Assets/Import_out/Uzil/API/Achv.cs
using UnityEngine;

using Uzil;
using Uzil.Achievement;

namespace UZAPI {

public class Achv {

	/*======================================Constructor==========================================*/

	/*=====================================Static Members========================================*/

	/*=====================================Static Funciton=======================================*/

	/*=========================================Members===========================================*/

	/*========================================Components=========================================*/

	/*==========================================Event============================================*/

	/*======================================Unity Function=======================================*/

	/*========================================Interface==========================================*/

	/** 是否 已解鎖 */
	public static bool? IsUnlock (string id) {
		AchvData data = AchvMgr.Inst().GetAchv(id);
		if (data == null) return null;
		return data.isUnlock;
	}

	/** 取得進度 */
	public static int? GetStat (string id) {
		AchvData data = AchvMgr.Inst().GetAchv(id);
		if (data == null) return null;
		return data.GetStat();
	}

	/** 設置 進度 */
	public static void SetStat (string id, int stat) {
		AchvMgr.Inst().SetStat(id, stat);
	}

	/** 設置 完成 */
	public static void SetDone (string id) {
		AchvMgr.Inst().SetDone(id);
	}

	/*=====================================Public Function=======================================*/

	/*===================================Protected Function======================================*/

	/*====================================Private Function=======================================*/

}

}